nixos/modules/security/permissions-wrappers/permissions-wrapper.c

```diff
@@ -0,0 +1,224 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <dirent.h>
+#include <assert.h>
+#include <errno.h>
+
+// Make sure assertions are not compiled out, we use them to codify
+// invariants about this program and we want it to fail fast and
+// loudly if they are violated.
+#undef NDEBUG
+
+extern char **environ;
+
+// The SOURCE_PROG and WRAPPER_DIR macros are supplied at compile time
+// for a security reason: So they cannot be changed at runtime.
+static char * sourceProg = SOURCE_PROG;
+static char * wrapperDir = WRAPPER_DIR;
+
+// Make sure we have the WRAPPER_TYPE macro specified at compile
+// time...
```

Comment:
Re: fetchability

I mean cache misses.

The old way is: you compile a single wrapper. it is probably available from Hydra due to NixOS tests, then you cp/echo things around, GCC and local compilation is not needed.

The new way: you compile a new binary per executable, and you want a wrapper binary for something like `smokeping` so Hydra lacks it, you now need local gcc for `nixos-rebuild`.